### PR TITLE
fix: only break loop when results is not null or empty

### PR DESCRIPTION
**Pull request by johevemi**
## References

* #0000

### CODE REVIEW

File: source/dotnet/domain-tests/Fixtures/AuthorizedClientFixtureOutput.cs [hunk 1]
```diff
@@ -88,7 +89,10 @@ private async Task<bool> WaitForCalculationToComplete(Guid calculationId)
                     var message = await _receiver.ReceiveMessageAsync();
                     if (message?.Body == null)
                     {
-                        messageHasValue = false;
+                        if (!results.IsNullOrEmpty())
```
Comment (JesperGomes) [inline source/dotnet/domain-tests/Fixtures/AuthorizedClientFixtureOutput.cs:92]: could this method be more compact like this?:

`
   
    
    var results = new List<CalculationResultCompleted>();
    using var cts = new CancellationTokenSource(TimeSpan.FromMinutes(5));
    while (!cts.Token.IsCancellationRequested)
    {
        var message = await _receiver.ReceiveMessageAsync();
        if (message?.Body == null && results.Any())
        {
            break;
        }

        var data = message.Body.ToArray();
        var result = CalculationResultCompleted.Parser.ParseFrom(data);
        if (result.BatchId == calculationId.ToString())
        {
            results.Add(result);
        }
    }

    return results;
`

Comment (JesperGomes) [reply source/dotnet/domain-tests/Fixtures/AuthorizedClientFixtureOutput.cs:92]: I might have missed some details. Just a suggestion.

Comment (johevemi) [reply source/dotnet/domain-tests/Fixtures/AuthorizedClientFixtureOutput.cs:92]: if the message.body.toarray() can handle that there might be nog message and parser too.

Comment (JesperGomes) [reply source/dotnet/domain-tests/Fixtures/AuthorizedClientFixtureOutput.cs:92]: Ah yes. Then the suggestion should instead be:
```
var results = new List<CalculationResultCompleted>();
using var cts = new CancellationTokenSource(TimeSpan.FromMinutes(5));
while (!cts.Token.IsCancellationRequested)
{
    var message = await _receiver.ReceiveMessageAsync();
    if (message?.Body == null)
    {
        if (results.Any())
             break;
   }
   else 
   {
      var data = message.Body.ToArray();
      var result = CalculationResultCompleted.Parser.ParseFrom(data);
      if (result.BatchId == calculationId.ToString())
      {
          results.Add(result);
      }
   }
}

return results;
```

I will approve and let it up to you.

**Review by JesperGomes — DISMISSED** [review]

File: source/dotnet/domain-tests/Fixtures/AuthorizedClientFixtureOutput.cs (hunk 1 above)
Comment (johevemi) [reply source/dotnet/domain-tests/Fixtures/AuthorizedClientFixtureOutput.cs:92]: that should work

**Review by JesperGomes — APPROVED** [review]

**Pull request merged**